.github/workflows/tox-tests.yml

```diff
@@ -11,64 +11,44 @@ on:
     tags:
       - '*'
   pull_request:
+  schedule:
+    # run every Monday at 6am UTC
+    - cron: '0 6 * * 1'
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
-env:
-  TOXARGS: '-v'
-
 jobs:
-  # Set up matrix to run tox tests across lists of os, python version, and tox environment
-  matrix_tests:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        # Github actions supports ubuntu, windows, and macos virtual environments:
-        # https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
-        #
-        # Only run on ubuntu by default, but can add other os's to the test matrix here.
-        # For example -- os: [ubuntu-latest, macos-latest, windows-latest]
-        include:
-          - os: ubuntu-latest
-            python: '3.10'
-            tox_env: 'py310-test-cov'
-          - os: ubuntu-latest
-            python: '3.11'
-            tox_env: 'py311-test'
-          - os: ubuntu-latest
-            python: '3.12'
-            tox_env: 'py312-test'
-          - os: macos-latest
-            python: '3.12'
-            tox_env: 'py312-test-devdeps'
-          - os: ubuntu-latest
-            python: '3.12'
-            tox_env: 'codestyle'
-          - os: ubuntu-latest
-            python: '3.10'
-            tox_env: 'py310-test-oldestdeps'
+  tests:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    secrets:
+      CODECOV_TOKEN: ${{ secrets.CODECOV }}
+    with:
+      submodules: false
+      coverage: ''
+      envs: |
+        - name: Codestyle
+          linux: codestyle
+
+        - name: Python 3.8 on Linux with oldest supported dependencies
+          linux: py38-test-alldeps-oldestdeps
+          toxargs: -v
+
+        - name: Python 3.9 on Windows with minimal dependencies
+          windows: py39-test
+          toxargs: -v
```

Comment:
we have discussed and decided to make 3.10 the minimum supported version. alas, that change is still staged in #182 which hasn't been merged yet.

Reply:
See https://github.com/astropy/specreduce/pull/202#discussion_r1418278247